backend/remote/remote.go

```diff
@@ -175,12 +177,21 @@ func (r *Remote) monitorPing(client *rpc.Client) error {
 	ticker := time.NewTicker(pingInveral)
 	defer ticker.Stop()
 
+	retry := 0
 	for {
 		select {
 		case <-r.closeChan:
 			return nil
 		case <-ticker.C:
-			if err := r.Ping(); err != nil {
+			if err := r.Ping(); err == nil {
+				retry = 0 // reset on success
+				break
+			} else {
+				if err == ErrPingTimeout && retry < pingRetries {
```

Comment:
What is the specific advantage of looking for `ErrPingTimeout`.  It we got something like a `net.Error` I think we would probably should not immediately fail.